client/lib/analytics/ad-tracking/google-analytics-4.ts

```diff
@@ -0,0 +1,36 @@
+import { GaPurchase } from '../utils/jetpack-cart-to-ga-purchase';
+import { GaItem } from '../utils/jetpack-product-to-ga-item';
+import { TRACKING_IDS } from './constants';
+
+// Ensure setup has run.
+import './setup';
+
+export function setup( params: Gtag.ConfigParams ) {
+	window.gtag( 'config', TRACKING_IDS.jetpackGoogleGA4Gtag, params );
+}
+
+export function fireJetpackEcommercePurchase( purchase: GaPurchase ) {
+	window.gtag( 'event', 'purchase', {
+		send_to: TRACKING_IDS.jetpackGoogleGA4Gtag,
+		...purchase,
+	} );
+}
+
+export function fireJetpackEcommerceAddToCart( item: GaItem ) {
```

Comment:
Thanks, I agree it can be more universal, but especially with TS typing - we should address it more properly once you're ready to move to GA4 with WPcom. Essentially, there is a lot more work to be done in the analytics library, I'd love to see it all typed (as calypso code) once we'll work more on it.